src/effects/VST/VSTEffect.cpp

```diff
@@ -804,10 +804,15 @@ VSTEffect::VSTEffect(const PluginPath & path, VSTEffect *master)
    mTimeInfo.timeSigDenominator = 4;
    mTimeInfo.flags = kVstTempoValid | kVstNanosValid;
 
-   // If we're a slave then go ahead a load immediately
+   // Get the constant data that are shared by all instances
+   // (e.g. entry point in the dll, vst version, no. of channels etc.)
+   LoadCommon();
+
+   // If we're a slave then go ahead and load immediately
    if (mMaster)
    {
-      Load();
+      if ( ! Load() )
+         Unload();
```

Comment:
So, construction of a `VSTEffect` may leave it in an unusable state if load fails.  Where should that failure be checked?  Where `make_unique<VSTEffect>` happens, and one place is `LoadPlugin`, just mentioned, the other in `VSTEffect::RealtimeAddProcessor`.

I don't yet see if you add a check there and return false, but I'm still marching through this commit sequence.